### PR TITLE
S3 ingestion can assume role

### PR DESCRIPTION
**Pull request by egor-ryashin**
### Description

This feature allows S3 ingestion to use the AssumeRole capability of AWS that helps with cross-account access.
It introduces 2 additional fields in S3InputSourceConfig (those are optional).

This PR has:
- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [x] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

File: docs/ingestion/native-batch.md [hunk 1]
```diff
@@ -862,6 +862,47 @@ Sample specs:
...
+...
+    "ioConfig": {
+      "type": "index_parallel",
+      "inputSource": {
+        "type": "s3",
+        "uris": ["s3://foo/bar/file.json", "s3://bar/foo/file2.json"],
+        "properties": {
+          "accessKeyId": "KLJ78979SDFdS2",
+          "secretAccessKey": "KLS89s98sKJHKJKJH8721lljkd",
+          "assumeRoleArn": "arn:aws:iam::2981002874992:role/role-s3"
```
Comment (nishantmonu51) [inline docs/ingestion/native-batch.md:895]: I believe above are some dummy IDs and roles and not some actual ones.

File: extensions-core/s3-extensions/src/main/java/org/apache/druid/data/input/s3/S3InputSourceConfig.java [hunk 2]
```diff
@@ -34,24 +34,44 @@
...
+  @Nullable
+  @JsonProperty
+  private String assumeRoleArn;
+  @Nullable
+  @JsonProperty
+  private String assumeRoleExternalId;
+  @JsonProperty
+  private PasswordProvider accessKeyId;
+  @JsonProperty
+  private PasswordProvider secretAccessKey;
```
Comment (nishantmonu51) [inline extensions-core/s3-extensions/src/main/java/org/apache/druid/data/input/s3/S3InputSourceConfig.java:46]: accessKeyId and SecretAccessKey are also nullable, Can we add annotation these as well.

File: docs/ingestion/native-batch.md (hunk 1 above)
Comment (egor-ryashin) [reply docs/ingestion/native-batch.md:895]: Yes. All fake.

File: extensions-core/s3-extensions/src/main/java/org/apache/druid/data/input/s3/S3InputSourceConfig.java (hunk 2 above)
Comment (egor-ryashin) [reply extensions-core/s3-extensions/src/main/java/org/apache/druid/data/input/s3/S3InputSourceConfig.java:44]: Added.

**Comment by egor-ryashin** [comment]
Fixed all except coverage. The code is integration with S3 API, writing mocks for it will effectively replace about 90% of code, which is unreasonable for testing.

**Review by nishantmonu51 — APPROVED** [review]
👍 , LGTM

**Review by pjain1 — APPROVED** [review]
some

File: extensions-core/s3-extensions/pom.xml [hunk 3]
```diff
@@ -115,8 +115,8 @@
     <dependency>
       <groupId>com.amazonaws</groupId>
       <artifactId>aws-java-sdk-sts</artifactId>
-      <scope>provided</scope>
```
Comment (clintropolis) [inline extensions-core/s3-extensions/pom.xml:118]: why this change?

Comment (egor-ryashin) [reply extensions-core/s3-extensions/pom.xml:118]: `aws-common` doesn't contain it, surprisingly Kinesis extension contains it, when Kinesis ext. is removed then the S3 ext. fails.
